### PR TITLE
Fixing broken nested pipeline

### PR DESCRIPTION
**Pull request by huguesBouvier**
This PR is intended to bring back the nested pipeline to basic running state.
- Took the opportunity to use super config for the lvl 4 and lvl5 and use apt-get instead of dpckg
- The lvl4 and lvl5 was fixed the right way, however I was lacking the bandwidth to fix the quickstart + Microsoft.Azure.Devices.Edge.Test the right way. Just made a quick fix to get it to work

https://dev.azure.com/msazure/One/_build/results?buildId=40961617&view=logs&j=e8afecc0-c259-5cc1-62e0-5c5d1c2a3db4&t=f401ce77-605a-59dc-34fb-0bcc6cca03c5

### CODE REVIEW

File: edge-modules/iotedge-diagnostics-dotnet/install.bat [hunk 1]
```diff
@@ -0,0 +1,5 @@
+ECHO OFF
+dotnet build
+dotnet publish
+docker build --no-cache -t huguesb.azurecr.io/azureiotedge-diagnostics:test --file ./docker/linux/amd64/Dockerfile --build-arg EXE_DIR=. ./bin/Debug/netcoreapp3.1/publish
+docker push huguesb.azurecr.io/azureiotedge-diagnostics:test
```
Comment (and-rewsmith) [inline edge-modules/iotedge-diagnostics-dotnet/install.bat:5]: We should remove this dev script right?

Comment (huguesBouvier) [reply edge-modules/iotedge-diagnostics-dotnet/install.bat:5]: oops thanks

File: e2e_deployment_files/nestededge_middleLayerBaseDeployment_amqp.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
           }
         },
         "modules": {
-          "iotedgeApiProxy": {
+          "IoTEdgeAPIProxy": {
```
Comment (and-rewsmith) [inline e2e_deployment_files/nestededge_middleLayerBaseDeployment_amqp.json:44]: My understanding is this rename won't have side effects.

**Review by and-rewsmith — APPROVED** [review]
I'm not super clear on why we did things the way we did it before. But the new flow looks good to me.

File: e2e_deployment_files/nestededge_middleLayerBaseDeployment_amqp.json (hunk 2 above)
Comment (huguesBouvier) [reply e2e_deployment_files/nestededge_middleLayerBaseDeployment_amqp.json:44]: It won't I am aligning with default name

File: scripts/linux/nested-edge-deploy-agent.sh [hunk 3]
```diff
@@ -14,133 +14,53 @@ function create_certificates() {
...
-    sudo chmod 644 /etc/aziot/identityd/config.toml
-    sudo cat /etc/aziot/identityd/config.toml
-
-    echo "Setup aziot-edged-principal.toml"
-    id_aziot=$(id -u iotedge)
-    sudo touch /etc/aziot/identityd/config.d/aziot-edged-principal.toml
-    echo "[[principal]]" | sudo tee  /etc/aziot/identityd/config.d/aziot-edged-principal.toml
-    echo "uid = ${id_aziot}" | sudo tee -a  /etc/aziot/identityd/config.d/aziot-edged-principal.toml
-    echo "name = \"aziot-edge\"" | sudo tee -a  /etc/aziot/identityd/config.d/aziot-edged-principal.toml
-    sudo chown "$(id -u aziotid):$(id -g aziotid)" /etc/aziot/identityd/config.d/aziot-edged-principal.toml
```
[the remark this answers is not in the document]
Comment (huguesBouvier) [reply scripts/linux/nested-edge-deploy-agent.sh:127]: Because that was the only way at the time :)

**Pull request merged**